File: edukate-checker/src/main/java/io/github/sanyavertolet/edukate/checker/services/ResultPublisher.java
package io.github.sanyavertolet.edukate.checker.services;

import io.github.sanyavertolet.edukate.common.checks.CheckResult;
import reactor.core.publisher.Mono;

public interface ResultPublisher {
    Mono<Void> publish(CheckResult result);
}

File: edukate-checker/src/main/kotlin/io/github/sanyavertolet/edukate/checker/services/ResultPublisher.kt
package io.github.sanyavertolet.edukate.checker.services

import io.github.sanyavertolet.edukate.common.checks.CheckResultMessage
import reactor.core.publisher.Mono

fun interface ResultPublisher {
    fun publish(result: CheckResultMessage): Mono<Void>
}
